File: questions and attempted answers/old/118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int):
        res = []
        if numRows == 0:
            return res
        pre = [1]
        res.append(pre)
        for i in range(0, numRows-1):
            middle = []
            length = len(pre)
            for j in range(length):
                if j == 0:
                    middle.append(pre[j])
                else:
                    middle.append(pre[j-1] + pre[j])

            middle.append(1)
            pre = middle
            res.append(middle)

        return res


if __name__ == "__main__":
    s = Solution()
    res = s.generate(0)
    print(res)
    

File: questions and attempted answers/old/118-pascals-triangle.cpp
/*
118. 杨辉三角
给定一个非负整数 numRows，生成杨辉三角的前 numRows 行。



在杨辉三角中，每个数是它左上方和右上方的数的和。

示例:

输入: 5
输出:
[
     [1],
    [1,1],
   [1,2,1],
  [1,3,3,1],
 [1,4,6,4,1]
]
*/

#include <vector>
#include <iomanip>
#include <iostream>
#include <stdio.h>
#include <typeinfo.h>
#include <string>

using namespace std;

class Solution
{
public:
    vector<vector<int>> generate(int numRows)
    {
        vector<vector<int>> res;
        vector<int> middle;
        middle.push_back(1);      
        res.push_back(middle);
        for (int i=0; i++; i<numRows)
        {
            res.push_back(middle);
            middle.push_back(1);
        }
        return res;
    }
};
